### DISCUSSION TITLE
default storage is overwritten on k3s restart

### DISCUSSION THREAD

**Discussion by n0cloud** [techno-tim/k3s-ansible#293] [Questions] [answered]
## Expected Behavior

After modifying the 'local-path' storage class to `default: false`, it should stay like that after k3s restart.

## Current Behavior
If I modify 'local-path' storage class, it's overwritten after k3s restart.

## Steps to Reproduce

1. patch the 'local-path' storage class with `storageclass.kubernetes.io/is-default-class: 'false'`
2. On the master node, restart k3s
3. manifests are recreated and default class is back to true

## Context (variables)
Running 3 ubuntu VM in proxmox with 1 master and 2 workers

Hardware:
a single 1235U nuc

## Possible Solution
The issue seem to be caused by k3s recreating the manifests files that we delete in the ansible.

For the moment, after running ansible the first time, I've set `--disable local-storage` flag on the master node to prevent recreating local-storage.yaml file.

Another solution seem to modify the manifest file instead of deleting it ( https://github.com/k3s-io/k3s/issues/3441 )

- [x] I've checked the [General Troubleshooting Guide](https://github.com/techno-tim/k3s-ansible/discussions/20)

**Comment by timothystewart6** [comment, marked as answer]
This is actually the default behavior of k3s.  Check here for the manifest: `/var/lib/rancher/k3s/server/manifests/`

All of the manifests here are reapplied each time the k3s service restarts.  I think you modify it there.

If you want to disable it when installing, add this arg  `--no-deploy local-storage`

Or you can delete the storage class as well as the manifest
